plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/ProjectWrapper.java

```diff
@@ -0,0 +1,115 @@
+package org.jboss.tools.openshift.internal.ui.models;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.openshift.restclient.model.IProject;
+import com.openshift.restclient.model.IResource;
+import com.openshift.restclient.model.IService;
+
+public class ProjectWrapper extends ResourceContainer<IProject, ConnectionWrapper> {
+	private AtomicReference<LoadingState> state = new AtomicReference<LoadingState>(LoadingState.INIT);
+
+	public ProjectWrapper(ConnectionWrapper parent, IProject resource) {
+		super(parent, resource);
+	}
+
+	public Collection<AbstractResourceWrapper<?, ?>> getResources() {
+		synchronized (childrenLock) {
+			return containedResources.values();

```

Comment:
The returned collection is not synchronized and using it may result in concurrency problems.